src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SubSourceLogic.cs

```diff
@@ -147,23 +201,68 @@ protected void ConfigureSubscription(Action<IImmutableSet<TopicPartition>> parti
 
             IPartitionEventHandler CreateRebalanceListener(IAutoSubscription subscription)
             {
-                return new PartitionEventHandlers.Chain(subscription.PartitionEventsHandler.GetOrElse(PartitionEventHandlers.Empty.Instance), new PartitionEventHandlers.AsyncCallbacks(subscription, SourceActor.Ref, partitionsAssignedCb,
-                    partitionsRevokedCb));
+                return new PartitionEventHandlers.Chain(
+                    subscription.PartitionEventsHandler.GetOrElse(PartitionEventHandlers.Empty.Instance),
+                    new PartitionEventHandlers.AsyncCallbacks(subscription, SourceActor.Ref, partitionsAssignedCb,
+                        partitionsRevokedCb));
             }
         }
-        
+
+        private class FlushMessagesOfRevokedPartitionsHandler : IPartitionEventHandler
```

Comment:
Because we can't use default interface implementations, I had to copy and paste this code over from the `SingleSourceLogic` file and make some minor changes to this, but basically this is what is responsible for filtering out the messages from revoked partitions.